### PR TITLE
Tickets/dm 31413

### PR DESCRIPTION
**Pull request by tribeiro**

### CODE REVIEW

File: python/lsst/ts/externalscripts/auxtel/build_pointing_model.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+            Azimuth of the grid position (in degrees).
+        elevation : `float`
+            Elevation of the grid position (in degrees).
+        """
+
+        try:
+            target = await self.atcs.find_target(
+                az=azimuth, el=elevation, mag_limit=self.config.magnitude_limit
+            )
+        except Exception:
```
Comment (patrickingraham) [inline python/lsst/ts/externalscripts/auxtel/build_pointing_model.py:234]: does it give any other information when it doesn't find a target? 
are there differnet exceptions that need to be handled differently? What it if can't connect or something?

File: python/lsst/ts/externalscripts/auxtel/build_pointing_model.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+    async def center_on_brightest_source(self):
+
+        self.latiss.rem.atarchiver.evt_imageInOODS.flush()
+
+        acquisition_image_ids = await self.latiss.take_engtest(
+            exptime=self.config.exposure_time, n=1
+        )
+
+        await self.latiss.rem.atarchiver.evt_imageInOODS.next(
```
Comment (patrickingraham) [inline python/lsst/ts/externalscripts/auxtel/build_pointing_model.py:256]: there is technically a race condition here.. but it probably doesn't matter.
i suppose the proper thing to do is verify the imageInOODS event matches the acquisition_image_ids

File: tests/auxtel/test_build_pointing_model.py [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+import numpy as np
+
+from lsst.ts.standardscripts import BaseScriptTestCase
+from lsst.ts.externalscripts import get_scripts_dir
+from lsst.ts.externalscripts.auxtel.build_pointing_model import BuildPointingModel
+from lsst.ts.observatory.control.utils.enums import RotType
+
+
+class TestBuildPointingModel(BaseScriptTestCase, unittest.IsolatedAsyncioTestCase):
+    def setUp(self) -> None:
```
Comment (patrickingraham) [inline tests/auxtel/test_build_pointing_model.py:35]: What does the function annotation of "None" do?

File: tests/auxtel/test_build_pointing_model.py [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+            unittest.mock.call(azimuth, elevation)
+            for azimuth, elevation in zip(
+                self.script.azimuth_grid, self.script.elevation_grid
+            )
+        ]
+        self.script.execute_grid.assert_has_awaits(calls)
+
+    async def test_execute_grid(self):
+
+        async with self.make_configured_dry_script():
```
Comment (patrickingraham) [inline tests/auxtel/test_build_pointing_model.py:129]: what is a dry script?

**Review by patrickingraham — CHANGES_REQUESTED** [review]
Looks really good.
Is there any reason to make this an integration test with real data? I don't think so but thought I'd ask....

File: Jenkinsfile [hunk 5]
```diff
@@ -1,12 +1,39 @@
...
+        work_branches = "${env.GIT_BRANCH} ${env.CHANGE_BRANCH} develop"
     }
 
     stages {
+        stage ('Cloning Repos') {
+            steps {
+                dir(env.WORKSPACE + '/ci/ts_externalscripts') {
+                    checkout scm
+                }
+                dir(env.WORKSPACE + '/ci/Spectractor') {
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply Jenkinsfile:22]: Yes, I don't remember which one uses it but either atmospec or rapid_analysis need it..

File: python/lsst/ts/externalscripts/auxtel/build_pointing_model.py [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+        self.get_image_timeout = 10.0
+
+        self.elevation_grid = np.array([])
+        self.azimuth_grid = np.array([])
+
+    @classmethod
+    def get_schema(cls):
+        yaml_schema = """
+$schema: http://json-schema/draft-07/schema#
+$id: https://github.com/lsst-ts/ts_standardscripts/auxtel/BuildPointingModelAndHexapodLUT.yaml
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply python/lsst/ts/externalscripts/auxtel/build_pointing_model.py:89]: Ops.. when I first started working on it I was going to do both in the script, but then I changed my mind.. good catch...

File: python/lsst/ts/externalscripts/auxtel/build_pointing_model.py [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+title: BuildPointingModelAndHexapodLUT v1
+description: Configuration for BuildPointingModelAndHexapodLUT.
+type: object
+properties:
+    nside:
+        type: integer
+        default: 3
+        description: Healpix nside parameter.
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply python/lsst/ts/externalscripts/auxtel/build_pointing_model.py:97]: Agree.. I improved the description.

File: python/lsst/ts/externalscripts/auxtel/build_pointing_model.py [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+
+        npix = hp.nside2npix(self.config.nside)
+
+        healpy_indices = np.arange(npix)
+
+        azimuth, elevation = hp.pix2ang(
+            nside=self.config.nside, ipix=healpy_indices, lonlat=True
+        )
+
+        mask_out_of_range = np.bitwise_and(
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply python/lsst/ts/externalscripts/auxtel/build_pointing_model.py:162]: how about `position_in_search_area_mask`?

File: python/lsst/ts/externalscripts/auxtel/build_pointing_model.py [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+
+        self.log.debug(f"Grid size: {self.grid_size}.")
+
+    @property
+    def grid_size(self):
+        return len(self.elevation_grid)
+
+    @property
+    def images_per_position(self):
+        return 2
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply python/lsst/ts/externalscripts/auxtel/build_pointing_model.py:185]: Yeah, I considered this for a bit. The current logic of the Script is not to iterate until it gets the start close enough to the center of the FoV but rather to center it once, take an exposure and register the position. Then later we can use the second image to refine the position. This allows us to run more quickly over the grid. 

We could change it to iterate until it reaches a certain limit, I am just not certain how much we would gain by that...

File: python/lsst/ts/externalscripts/auxtel/build_pointing_model.py [hunk 10]
```diff
@@ -0,0 +1,290 @@
...
+
+    @property
+    def camera_readout_time(self):
+        return 2.0
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply python/lsst/ts/externalscripts/auxtel/build_pointing_model.py:189]: I agree.. it is a stop-gap until we get that sorted out.. I think that eventually the camera should publish this values and we grab it from there... but I agree the next step would be to add this to latiss_constants.

File: python/lsst/ts/externalscripts/auxtel/build_pointing_model.py [hunk 11]
```diff
@@ -0,0 +1,290 @@
...
+
+    @property
+    def estimated_average_slew_time(self):
+        """The estimated average slew time considers a slew speed of 1 deg/sec.
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply python/lsst/ts/externalscripts/auxtel/build_pointing_model.py:193]: Maybe the slew speed could go there, but this method is slightly trickier, it considers what is the resolution of the pointing grid and then how many paintings there are to estimate the average slew time.

File: python/lsst/ts/externalscripts/auxtel/build_pointing_model.py (hunk 1 above)
Comment (tribeiro) [reply python/lsst/ts/externalscripts/auxtel/build_pointing_model.py:234]: Yes, indeed it does issue different kinds of execeptions for different kinds of errors. I haven't figured out which is which tough.

File: python/lsst/ts/externalscripts/auxtel/build_pointing_model.py (hunk 2 above)
Comment (tribeiro) [reply python/lsst/ts/externalscripts/auxtel/build_pointing_model.py:256]: uhm... I see.. maybe we should add something to `LATISS` to deal with that.. I've seen this more than once and you are right, we may want to check that the `imageInOODS` event is for the correct image id.

File: tests/auxtel/test_build_pointing_model.py (hunk 3 above)
Comment (tribeiro) [reply tests/auxtel/test_build_pointing_model.py:35]: That is python type annotation.. just specifying that the `setUp` method does not return anything (or return `None`, which is the same thing).

File: python/lsst/ts/externalscripts/auxtel/build_pointing_model.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (patrickingraham) [reply python/lsst/ts/externalscripts/auxtel/build_pointing_model.py:162]: that  works, but `search_area_mask` works too (and is shorter). Your call.

File: python/lsst/ts/externalscripts/auxtel/build_pointing_model.py (hunk 2 above)
Comment (patrickingraham) [reply python/lsst/ts/externalscripts/auxtel/build_pointing_model.py:273]: So the idea was to refactor this stuff when we switch to gen3... which was supposed to be this week...

**Review by patrickingraham — APPROVED** [review]
Made a couple minor comments.
I think the LATISS class needs some tweaks... but that should be a different ticket.

**Pull request merged**